src/type/definition.js

```diff
@@ -649,17 +649,17 @@ export class GraphQLObjectType {
   extensionASTNodes: ?$ReadOnlyArray<ObjectTypeExtensionNode>;
   isTypeOf: ?GraphQLIsTypeOfFn<*, *>;
 
-  _typeConfig: GraphQLObjectTypeConfig<*, *>;
```

Comment:
The memory leak happens because `_typeConfig` contains `field` and `interfaces` closures that depend on some internal arrays + in case of `extendSchema` it also depends on the previous schema.